src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -1098,6 +1098,15 @@ export class CellModel extends Disposable implements ICellModel {
 		this.fireOutputsChanged(false);
 	}
 
+	public processEdits(edits: ICellEdit[]): void {
+		// Wait for any in-progress operations to complete before doing any edits so that we don't hit any race conditions with updating cell outputs
```

Comment:
Isn't this going to break being able to update outputs as the cell is running though? That was specifically something I made sure could happen - the idea being that a cell may want to send partial/streamed results while it's executing.

Reply:
Do you have an example of code that would stream output like that?

Reply:
Not anymore - and this was for a custom test kernel I made when I was implementing this stuff. It hinges around using [appendOutput](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L13277) on the Notebook extension APIs to modify the output of a cell during execution.

Reply:
I'll take a look and see how this change impacts that

Reply:
So the outputs still show up correctly when using append, but they all get displayed at once when the execution is done. I'll change this to fix that.